Server/src/routes/init.route.ts

```diff
@@ -199,11 +199,22 @@ export class InitRoute {
 	private async getAppPoisFromDbPoi(pois: POI[]): Promise<PointOfInterest[]> {
 		const apiPois: PointOfInterest[] = []
 		for (const poi of pois) {
-			const type: number = poi.typeId
+			const type: POIType | null = await POIService.getPOITypeById(poi.typeId)
 			if (!type) {
 				logger.error(`Could not determine type of poi with id ${poi.uid}`)
 				continue
 			}
+			const poiIcon: number = Number.parseInt(type.icon)
+			if (!Number.isInteger(poiIcon)) {
+				logger.error(`Icon of type with id ${type.uid} is not an integer.`)
+				continue
+			}
+			// Check if the icon number is a member of the enum.
+			if (!(poiIcon in POITypeIcon)) {
```

Comment:
As an `enum` is compiled down to an object, this should check -- using the backwards mapping -- if the number is a value assigned to an enum member.